src/main/java/org/frankframework/frankdoc/AttributeReuseManager.java

```diff
@@ -0,0 +1,180 @@
+You may obtain a copy of the License at 
+
+    http://www.apache.org/licenses/LICENSE-2.0 
+
+Unless required by applicable law or agreed to in writing, software 
+distributed under the License is distributed on an "AS IS" BASIS, 
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. 
+See the License for the specific language governing permissions and 
+limitations under the License. 
+*/
+package org.frankframework.frankdoc;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import org.frankframework.frankdoc.model.FrankAttribute;
+import org.frankframework.frankdoc.util.XmlBuilder;
+
+class AttributeReuseManager {
+	private static class AttributeToBuild {
+		FrankAttribute attribute;
+		XmlBuilder group;
+		boolean reused = true;
+		String targetName;
+
+		AttributeToBuild(FrankAttribute attribute, XmlBuilder group, String targetName) {
+			this.attribute = attribute;
+			this.group = group;
+			this.targetName = targetName;
+		}
+	}
+
+	private static class AttributeToInsert {
+		XmlBuilder attributeBuilder;
+		XmlBuilder parentBuilder;
+
+		AttributeToInsert(XmlBuilder attributeBuilder, XmlBuilder parentBuilder) {
+			this.attributeBuilder = attributeBuilder;
+			this.parentBuilder = parentBuilder;
+		}
+	}
+
+	private static class AttributeToBuildGroup {
+		private List<AttributeToBuild> itemsThatShareAttribute = new ArrayList<>();
+
+		AttributeToBuildGroup(AttributeToBuild first) {
+			itemsThatShareAttribute.add(first);
+		}
+
+		void add(AttributeToBuild item) {
+			if(item.attribute != itemsThatShareAttribute.get(0).attribute) {
+				throw new IllegalArgumentException("Programming error, see stack trace");
+			}
+			itemsThatShareAttribute.add(item);
+		}
+
+		boolean isAttributeHasUniqueGroup() {
+			return itemsThatShareAttribute.size() == 1;
+		}
+
+		String getAttributeName() {
+			return itemsThatShareAttribute.get(0).attribute.getName();
+		}
+
+		FrankAttribute getCommonFrankAttribute() {
+			return itemsThatShareAttribute.iterator().next().attribute;
+		}
+
+		void setNoItemReuses() {
+			itemsThatShareAttribute.stream().forEach(item -> item.reused = false);
+		}
+	}
+
+	private static class AttributeToBuildGroupGroup {
+		private Map<FrankAttribute, AttributeToBuildGroup> groupsSharingAttributeName = new LinkedHashMap<>();
+
+		AttributeToBuildGroupGroup(AttributeToBuild first) {
+			AttributeToBuildGroup group = new AttributeToBuildGroup(first);
+			groupsSharingAttributeName.put(first.attribute, group);
+		}
+
+		void add(AttributeToBuild item) {
+			if(! item.attribute.getName().equals(groupsSharingAttributeName.values().iterator().next().getAttributeName())) {
+				throw new IllegalArgumentException("Programming error, see stack trace");
+			}
+			if(groupsSharingAttributeName.containsKey(item.attribute)) {
```

Comment:
Klopt het dat je hier bepaalt waar het attribuut gecreeerd moet worden of hergebruikt kan worden?
Kan je dat (of wat er wel gebeurd) in een commentaarregel zetten?

Reply:
Die logica zit niet in een subclass maar in AttributeReuseManager zelf. Ik heb twee subclasses naar onderen geschoven zodat de logica in AttributeReuseManager meer nadruk krijg. Ik heb ook JavaDoc commentaren bij de methods in AttributeReuseManager gezet die van buitenaf worden aangeroepen. Zo wordt het oog naar het beginpunt van het algoritme getrokken in plaats van naar minder belangrijke details.